smart_contract/executor/src/tokens.rs

```diff
@@ -0,0 +1,478 @@
+//! Definition of Iroha default permission tokens
+#![allow(missing_docs, clippy::missing_errors_doc)]
+
+use alloc::{borrow::ToOwned, format, string::String, vec::Vec};
+
+use iroha_smart_contract::data_model::{prelude::*, executor::Result};
+use iroha_executor_derive::ValidateGrantRevoke;
+
+use crate::permission::{self, Token as _};
+
+/// Declare token types of current module. Use it with a full path to the token.
+/// Used to iterate over tokens to validate `Grant` and `Revoke` instructions.
+///
+///
+/// TODO: Replace with procedural macro. Example:
+/// ```
+/// mod tokens {
+///     use std::borrow::ToOwned;
+///
+///     use iroha_schema::IntoSchema;
+///     use iroha_executor_derive::{Token, ValidateGrantRevoke};
+///     use serde::{Deserialize, Serialize};
+///
+///     #[derive(Clone, PartialEq, Deserialize, Serialize, IntoSchema, Token, ValidateGrantRevoke)]
+///     #[validate(iroha_executor::permission::OnlyGenesis)]
+///     pub struct MyToken;
+/// }
+/// ```
+macro_rules! declare_tokens {
```

Comment:
I think this might be usefull inside `lib.rs` so that users might also use it without affiliation with default tokens